Geometry_Engine/Modify/RemoveLeastSignificantVertices.cs

```diff
@@ -41,43 +41,69 @@ public static partial class Modify
         [Input("angleTolerance", "The tolerance of the angle that defines a straight line. Default is set to the value defined by BH.oM.Geometry.Tolerance.Angle")]
         [Output("polyline", "The cleaned polyline")]
         public static Polyline RemoveLeastSignificantVertices(this Polyline polyline, double angleTolerance = Tolerance.Angle)
-        {
-            //This method is for closed polylines only at the moment
-            if (!polyline.IsClosed())
-            {
-                BH.Engine.Reflection.Compute.RecordError("The RemoveLeastSignificantVertices method is only for closed polylines and the input polyline is not closed.");
-                return polyline;
-            }
-
+        {            
             List<Point> pnts = polyline.DiscontinuityPoints();
+            List<Point> originalPnts = polyline.DiscontinuityPoints();
 
             if (pnts.Count < 3)
                 return polyline; //If there's only two points here then this method isn't necessary
 
             int startIndex = 0;
-            while (startIndex < pnts.Count)
+
+            if (polyline.IsClosed())
+            {
+                while (startIndex < pnts.Count)
+                {
+                    Point first = pnts[startIndex];
+                    Point second = pnts[(startIndex + 1) % pnts.Count];
+                    Point third = pnts[(startIndex + 2) % pnts.Count];
+
+                    if (first.Angle(second, third) <= angleTolerance)
+                        pnts.RemoveAt((startIndex + 1) % pnts.Count);  //Delete the second point from the list, it's not necessary
+                    else
+                        startIndex++; //Move onto the next point
+                }
+            }
+            else 
             {
-                Point first = pnts[startIndex];
-                Point second = pnts[(startIndex + 1) % pnts.Count];
-                Point third = pnts[(startIndex + 2) % pnts.Count];
+                while ((startIndex) < (pnts.Count - 1))
```

Comment:
I do not believe that any of this `if` `else` is necessary - I believe the algorithm will work quite happily as it was for non-closed polylines as it did for closed polylines.

Alternatively, if the `if` `else` is necessary, then the `else` needs modifying to work with non-closed polylines, because currently that algorithm will loop to the beginning of the polyline unnecessarily (unnecessarily because the polyline is not closed so we don't need to loop to the beginning again).